contracts/LimitOrder.sol

```diff
@@ -78,11 +79,19 @@ contract LimitOrder is ILimitOrder, BaseLibEIP712, SignatureValidator, Reentranc
         _;
     }
 
-    function transferOwnership(address _newOperator) external onlyOperator {
+    function nominateNewOperator(address _newOperator) external onlyOperator {
         require(_newOperator != address(0), "LimitOrder: operator can not be zero address");
-        operator = _newOperator;
+        nominatedOperator = _newOperator;
 
-        emit TransferOwnership(_newOperator);
+        emit OperatorNominated(_newOperator);
+    }
+
+    function acceptOwnership() external {
+        require(msg.sender == nominatedOperator, "LimitOrder: not nominated");
+        emit OperatorChanged(operator, nominatedOperator);
+
+        operator = nominatedOperator;
+        nominatedOperator = address(0);
```

Comment:
Not sure if we need time lock here as well? Although splitting into two steps, if no time lock, user is still in the risky situation.

Reply:
I think this is mainly to prevent operator error.

Reply:
Yeah, this prevent incorrect operator to be assigned.